### DISCUSSION TITLE
Adding new hashsets using NSRL format

### DISCUSSION THREAD

**Discussion by tuliorosa** [sepinf-inc/IPED#368] [Q&A] [answered]
Estou adicionando novos hash usando o -importkff
Mas percebi que é obrigatório ter o MD5 e o SHA-1

Estou pensando em colocar um hash fake no MD5 e o real no SHA-1, funcionária ?
O IPED usa preferencialmente um dos 2 ? ou faz uso dos 2 ?

Automatic translation:

I'm adding new hashes using -importkff
But I realized that it is mandatory to have MD5 and SHA-1

I'm thinking of putting a fake hash on MD5 and the real one on SHA-1, employee?
Does IPED preferably use one of the 2? or does it use the 2?

**Comment by lfcnassif** [comment, marked as answer]
Please post in english.

Just one of them is used, the one enabled in IpedConfig.txt. If both are enabled, md5 is used. If you just import sha-1 and use a fake md5, you should disable md5 and enabled sha-1 in that file, or hash look up will not work. Leaving md5 column empty does not work?

**Reply by tuliorosa** [reply]
There is a validation of the size of MD5 and SHA1.
You need to add both

**Comment by tuliorosa** [comment]
Imported MD5, and the fake SHA1.
Initially kff.db.p was 1.26 GB, after importing it was 3.71 GB.
When executing IPED, it presented errors.
See the attached LOG.
[IPED-2021-02-18-07-54-22.log](https://github.com/sepinf-inc/IPED/files/6001933/IPED-2021-02-18-07-54-22.log)

**Reply by lfcnassif** [reply]
Did you create a NSRLProd.txt file side by side with NSRLFile.txt before importing? Both files are mandatory.

**Reply by lfcnassif** [reply]
Also take a look at https://github.com/sepinf-inc/IPED/discussions/73

**Reply by tuliorosa** [reply]
Yes ... I created the NSRLProd.txt and NSRLFile.txt files, I had no problem importing.
After importing when I went to use the IPED it showed errors.

**Reply by tuliorosa** [reply]
I initially imported the hashes from the site "https://www.nist.gov/itl/ssd/software-quality-group/national-software-reference-library-nsrl"
After I imported the hashes from the site "https://www.hashsets.com/".

When executing the IPED of the error ... if only the NIST hashes are imported, the error is not present.


![image](https://user-images.githubusercontent.com/50704600/108388840-d1718e00-71ed-11eb-94da-db95559ba681.png)

**Reply by lfcnassif** [reply]
Possibly the hashes imported from the second site are with some issue. My guess is that NSRLFile.txt is pointing to some ProductCode not present in NSRLProd.txt, this is not verified when importing currently.

**Reply by tuliorosa** [reply]
All NSRLFile.txt Product Codes are present in NSRLProd.txt.
I'm going to do some tests.

**Reply by lfcnassif** [reply]
If you can provide a small hashset in NSRL format that reproduces the error, I can take a look.

**Reply by tuliorosa** [reply]
The problem was in ProducCode, it cannot be -2.
I changed the ProducCode to 999992, and solved it.

**Reply by lfcnassif** [reply]
Thanks for digging into it.
